Assets/Scripts/Interaction Scripts/ControllerInteractions.cs

```diff
@@ -500,7 +503,7 @@ private Waypoint CreateWaypoint(Vector3 groundPoint)
         /// </summary>
         public void UndoAndDeleteWaypoints()
         {
-            Drone currentlySelectedDrone = WorldProperties.selectedDrone;
+            Drone currentlySelectedDrone = WorldProperties.GetSelectedDrone();
 
             // Make sure the currently selected drone has waypoints
             if (currentlySelectedDrone.waypoints != null && currentlySelectedDrone.waypoints.Count > 0)
```

Comment:
this is good code but just curious, realistically speaking, when would waypoints == null? I thought we initialized with 1 takeoff waypoint?

Reply:
in case the user presses undo before placing